apps/meteor/server/models/raw/LivechatRooms.ts

```diff
@@ -1992,6 +1992,32 @@ export class LivechatRoomsRaw extends BaseRaw<IOmnichannelRoom> implements ILive
 		return this.find(query, options);
 	}
 
+	async findNewestByVisitorIdOrToken<T extends Document = IOmnichannelRoom>(
+		visitorId: string,
+		visitorToken: string,
+		options: Omit<FindOptions<IOmnichannelRoom>, 'sort' | 'limit'> = {},
+	): Promise<T | null> {
+		const query: Filter<IOmnichannelRoom> = {
+			t: 'l',
+			$or: [
+				{
+					'v._id': visitorId,
+				},
+				{
+					'v.token': visitorToken,
+				},
+			],
+		};
+
+		const cursor = this.find<T>(query, {
```

Comment:
why not use a `findOne`?

Reply:
I wasn't sure if findOne would consider the `sort`

Reply:
It does, @pierre-lehnen-rc . I'll use a `findOne` instead